src/main/java/app/Main.java

```diff
@@ -62,8 +62,8 @@ public static void saveAllMoviesInDB(String countryName) {
 
         MovieDAO movieDAO = new MovieDAO(emf);
         for (MovieDTO movieDTO : movies) {
-            Set<CastMemberDTO> castMembers = movieService.getCastMembersByMovieId(movieDTO.getId().intValue());
-            movieDTO.setCast(castMembers.stream().toList());
+            Set<CastMemberDTO> castMembers = movieService.getCastMembersByMovieId(movieDTO.getId());
+            movieDTO.setCast(new HashSet<>(castMembers));
```

Comment:
There's no need to convert `castMembers` to a HashSet as it is already a Set

Reply:
Changed.